backends/dpdk/dpdkUtils.cpp

```diff
@@ -151,15 +151,13 @@ bool reservedNames(P4::ReferenceMap *refMap, const std::vector<cstring> &names,
     return true;
 }
 
-/// Update bitwidth of Metadata fields to 32 or 64 bits if it 8-bit aligned.
+/// The DPDK pipeline requirement is all header/metadata fields to be multiple of 8bits.
```

Comment:
The BMv2 back end has a pass that throws an error when a type does not have the correct bit-width. Do the same thing here? Also isn't this restriction for total width? What if you have a 4-bit version field like in the IPv4 header?

Reply:
I discussed with Cristian and he told that all header/metadata/variables should be multiple of 8bits. There may be sub fields within 8bit like IPV4 headers. That compiler handles as the 8bit are finally covered and all headers are always multiple of 8bits. 

If user creates structure, for example with 3 fields as 24bits, 3bits, 32bits, here the 3bits field is not supported. For this, Cristian advises to issue error. But in compiler, i need some more time to do that. So, as of now, i am rounding to nearest 8bit multiples if fields are odd like in this example. In case if fields are 8bit, 2bits, 3bits, 1bit, 2bits, here compiler can pack  the last 4fields into one 8bit field and will not round them individually.

I will address the error issuance separately. Currently, rounding to nearest 8bits is fine.

Reply:
>I will address the error issuance separately. Currently, rounding to nearest 8bits is fine.

Will this lead to incorrect processing? Or does it just waste resources? If it is the former it is a bug and should not be merged. 

The BMv2 back end checks for headers like this fwiw: https://github.com/p4lang/p4c/blob/main/backends/bmv2/common/header.cpp#L257